docs/madr/decisions/046-policy-on-namespace.md

```diff
@@ -0,0 +1,456 @@
+# Policy On Namespace
+
+- Status: accepted
+
+## Context and Problem Statement
+
+Kuma 2.6.0 introduced a new feature allowing users to apply policies on Zone CP.
+This change unblocks us to introduce a namespace support for policies.
+The CRDs for new policies are already namespace-scoped, but they can be applied only on the `kuma-system` namespace.
+
+## Considered Options
+
+- allow applying new policies to the custom namespaces
+- do nothing
+
+## Decision Outcome
+
+- allow applying new policies to the custom namespaces
+
+## Pros and Cons of the applying policies on custom namespaces
+
+* Good, because allows using Kubernetes RBAC
+* Good, because provides more predictable behaviour as namespace-scoped policy affects only workloads in the same
+  namespace
+* Good, because provides more Kubernetes-native UX
+* Bad, because adds complexity for users
+    * what's the right namespace for the policy?
+    * how cross-policy refs work?
+    * how policy order works?
+
+## Pros and Cons of doing nothing
+
+* Good, because we have time to work on other features
+* Bad, because cluster operator has to grant write permissions to `kuma-system` namespace to everyone who works with
+  policies
+* Bad, because low isolation between teams (team-a can unintentionally break polices of the team-b)
+
+## Implementation
+
+### Namespace-scoped policies and multizone
+
+Namespace-scoped policies can be applied to a custom namespace only on zones as there are simply no DPPs and custom
+namespaces on Global.
+
+After a user applied a policy on zone's custom namespace, it automatically gets a `k8s.kuma.io/namespace` label.
+The label is part of the hash suffix when policy is synced to global.
```

Comment:
just to double check. There are no plans to convert targetRef implicitly or explicitly for example `targetRef:Mesh` in a namespace x to `targetRef:MeshSubset,namespace:x`. We are planning to add additional logic in policy matching. Is that correct?